modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java

```diff
@@ -312,4 +312,13 @@ public synchronized void shutdown() {
         start.set(false);
         workers.shutdown();
     }
+
+    /**
+     * Reasoning behind only returning iterator is that we
+     * don't want other parties to unsafely modify the mapping
+     * in an unsafe fashion
+     */
+    public Iterator<Map.Entry<Integer, PeerState>> getPeerStatesIterator() {
```

Comment:
Instead of leaking the iterator (safe, since won't yield concurrent modification exceptions, but behaviour undefined (see https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html)). 

Maybe better to just return a copy using copy constructors:
```
Set<K> keySetCopy = new HashSet<>(map.keySet());
List<V> valuesCopy = new ArrayList<>(map.values());
```